File: app/src/main/java/com/dvor/my/mydvor/message/MessageAdapter.java
package com.dvor.my.mydvor.message;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.dvor.my.mydvor.utils.DataConverter;
import com.dvor.my.mydvor.R;
import com.dvor.my.mydvor.data.Message;

import java.util.List;

public class MessageAdapter extends ArrayAdapter<Message> {

    private LayoutInflater inflater;
    private int layout;
    private List<Message> messages;

    public MessageAdapter(Context context, int resource, List<Message> messages) {
        super(context, resource, messages);
        this.messages = messages;
        this.layout = resource;
        this.inflater = LayoutInflater.from(context);
    }
    public View getView(int position, View convertView, ViewGroup parent) {

        View view=inflater.inflate(this.layout, parent, false);


        TextView titleView = view.findViewById(R.id.title);
        TextView textView = view.findViewById(R.id.text);
        TextView dataView = view.findViewById(R.id.data);
        Message message = messages.get(position);

        titleView.setText(message.getTitle());
        textView.setText(message.getText());
        dataView.setText(DataConverter.convert(message.getData()));
        return view;
    }
}

File: app/src/main/java/com/dvor/my/mydvor/message/MessageAdapter.kt
package com.dvor.my.mydvor.message

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import android.widget.Toast
import androidx.fragment.app.FragmentActivity
import com.dvor.my.mydvor.R
import com.dvor.my.mydvor.data.Message
import com.dvor.my.mydvor.utils.DateConverter

class MessageAdapter(context: FragmentActivity?, private val layout: Int, private val messages: List<Message>) : ArrayAdapter<Message>(context!!, layout, messages) {

    private val inflater: LayoutInflater = LayoutInflater.from(context)

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {

        val view = inflater.inflate(this.layout, parent, false)

        val titleView = view.findViewById<TextView>(R.id.title)
        val textView = view.findViewById<TextView>(R.id.text)
        val dataView = view.findViewById<TextView>(R.id.data)
        val message = messages[position]

        titleView.text = message.title
        textView.text = message.text
        dataView.text = DateConverter.convert(message.date.toString())
        dataView.setOnClickListener {
            Toast.makeText(context, message.date.toString(), Toast.LENGTH_SHORT).show()
        }
        return view
    }
}